### ISSUE TITLE
Missing path prop on <Route/> does not cause warning or error

### ISSUE THREAD

**Issue by jcheroske** [ReactTraining/react-router#5487]
## Version
4.2.2

## Steps to reproduce
Make a simple mistake while using the `Route` component, such as:
```
<Route to='/some/path' component={Foo} />
```

## Expected Behavior
A PropTypes warning for the missing `path` prop at least. Recommendation: use the `prop-types-exact` package to enforce allowing only the props listed in the component's `propTypes`.

## Actual Behavior
No error or warning is generated.

**Comment by pshrmn** [comment]
As noted in the [documentation](https://reacttraining.com/react-router/core/api/Route/path-string), a `<Route>` with no `path` _is_ valid and matches every route.

**Comment by jcheroske** [comment]
Can the props be checked, and unknown props then generate a warning? I and
others have accidentally named what is supposed to be the `path` prop `to`
which resulted in a difficult to track down bug.

**Comment by pshrmn** [comment]
The `<Route>` could probably have a `warning` call in its `componentWillMount` that logs a warning when the route has a `to` prop.

Your best bet for getting something like this included would be to open a PR. There are already a couple of `warning` call expressions in `cWM` so it should be fairly easy to implement (although a word of warning, the first param passed to the `warning` function is probably the opposite of what you think it should be. That has tripped me up multiple times).

**Comment by jcheroske** [comment]
I forked and tried to simply add the prop-types-exact package and then use
it in the Route component, but I ran into build issues. It seemed like
babel, during the build of rr4, was trying to transpile the node_modules
folder. I couldn't get the build to stop blowing up. Is there a secret to
adding a new third-party package? I would be happy to add exact prop
checking to the entire library, but after fussing with the build for an
hour I'm at an impasse.

**Comment by jcheroske** [comment]
I forked and tried to simply add the `prop-types-exact` package and then use
it in the Route component, but I ran into build issues. It seemed like
babel, during the build of rr4, was trying to transpile the node_modules
folder. I couldn't get the build to stop blowing up. Is there a secret to
adding a new third-party package? I would be happy to add exact prop
checking to the entire library, but after fussing with the build for an
hour I'm at an impasse.
